src/main/java/org/opentripplanner/model/StopPattern.java

```diff
@@ -79,19 +79,19 @@ int findStopPosition(StopLocation stop) {
     }
 
     int findBoardingPosition(StopLocation stop) {
-        return findStopPosition(0, stops.length-1, (s) -> s == stop, stop);
+        return findStopPosition(0, stops.length - 1, (s) -> s == stop);
     }
 
     int findAlightPosition(StopLocation stop) {
-        return findStopPosition(1, stops.length, (s) -> s == stop, stop);
+        return findStopPosition(1, stops.length, (s) -> s == stop);
     }
 
     int findBoardingPosition(Station station) {
-        return findStopPosition(0, stops.length-1, station::includes, station);
+        return findStopPosition(0, stops.length - 1, station::includes);
     }
 
     int findAlightPosition(Station station) {
-        return findStopPosition(1, stops.length, station::includes, station);
+        return findStopPosition(1, stops.length, station::includes);
```

Comment:
Would you mind adding the following JavaDoc to the `TripPattern` methods, the TripPattern is the public API and delegates to StopPattern(encapsulated in TripPattern):
```
    /**
     * Find the first stop position in pattern matching the given {@code stop}. The search start at
     * position {@code 0}. Return a negative number if not found.
     * Use {@link #findAlightStopPositionInPattern(StopLocation)} or 
     * {@link #findBoardingStopPositionInPattern(StopLocation)} if possible. 
     */
    public int findStopPosition(StopLocation stop) {

    /**
     * Find the first stop position in pattern matching the given {@code station} where it is
     * allowed to board. The search start at position {@code 0}. Return a negative number if not
     * found.
     */
    public int findBoardingStopPositionInPattern(Station station) {

    /**
     * Find the first stop position in pattern matching the given {@code station} where it is 
     * allowed to alight. The search start at position {@code 1}. Return a negative number if not
     * found.
     */
    public int findAlightStopPositionInPattern(Station station) {

    /**
     * Find the first stop position in pattern matching the given {@code stop} where it is allowed
     * to board. The search start at position {@code 0}. Return a negative number if not found.
     */
    public int findBoardingStopPositionInPattern(StopLocation stop) {

    /**
     * Find the first stop position in pattern matching the given {@code stop} where it is allowed
     * to alight. The search start at position {@code 1}. Return a negative number if not found.
     */
    public int findAlightStopPositionInPattern(StopLocation stop) {
```

Reply:
Added in https://github.com/opentripplanner/OpenTripPlanner/pull/3868/commits/bab6818d9fec41ca2b415526350df5515f5911cb